[Lim Jia Yan] iP

Jiayan-Lim opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE!**

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)
___

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Jiayan-Lim committed:
Lvl1: Implement Greet, Echo, Exit

Jiayan-Lim committed:
Lvl2: Implement add, list

Jiayan-Lim committed:
Lvl3: Implement mark, unmark

Jiayan-Lim committed:
Lvl4: Implement ToDos, Events, Deadlines

Jiayan-Lim committed:
A-TextUiTesting: Automated Text UI Testing

Jiayan-Lim committed:
fixed error in list function

Jiayan-Lim committed:
Lvl5: Handle error

Jiayan-Lim committed:
Lvl6: Implement delete

Jiayan-Lim committed:
Duke class: save data in hard disk

Task are only save as variable in Duke class.

Save data in hard disk can prevent data loss.

Save the tasks in the hard disk automatically whenever the task list
changes. Load the data from the hard disk when Duke starts up.

Jiayan-Lim committed:
Add data and time to events.

Duke treats date and time as string.

Duke should treat it as LocalDateTime class to perform in a more
meaningful ways.

Change it from String class to LocalDateTime class.

Jiayan-Lim committed:
Merge branch 'branch-Level-7'

Jiayan-Lim committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadlines.java
#	src/main/java/Duke.java
#	src/main/java/Events.java

Jiayan-Lim committed:
Merge branch 'add-gradle-support' of https://github.com/Jiayan-Lim/ip

Jiayan-Lim committed:
Add Junit Test

Jiayan-Lim committed:
Use More OOP

All command are deal within Duke.java.

Build more meaningful class to push the design more towards the OOP
approach.

Implement additional 5 class(UI, Storage, Parser, TaskList).

Jiayan-Lim committed:
Organize into Packages

Jiayan-Lim committed:
Mark,Unmark,Delete :Debug found error

it delete the wrong line and print out details of wrong line.

Fixed it by debugging it.

Jiayan-Lim committed:
JAR file

Jiayan-Lim committed:
Implement JavaDoc

Jiayan-Lim committed:
Follow the Coding Standard

Jiayan-Lim committed:
Implement Find

Jiayan-Lim committed:
Merge branch 'branch-A-JavaDoc'

Jiayan-Lim committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java

Jiayan-Lim committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/AddCommand.java
#	src/main/java/duke/Command.java
#	src/main/java/duke/Deadlines.java
#	src/main/java/duke/DeleteCommand.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Events.java
#	src/main/java/duke/ExitCommand.java
#	src/main/java/duke/ListCommand.java
#	src/main/java/duke/MarkCommand.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/ToDos.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/UnmarkCommand.java

Jiayan-Lim committed:
Use CheckStyle

Jiayan-Lim committed:
Update build.gradle for checkstyle

Jiayan-Lim committed:
Add GUI to Duke

Jiayan-Lim committed:
Merge branch 'branch-Level-10'

Jiayan-Lim committed:
Update checkstyle

leeyi45 commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,47 @@
+ * Deadline type of task
+ */
+public class Deadlines extends Task {
+    protected String ddl;
```
Some people consider even protected fields "exposed", so getters and setters might be more appropriate

Jiayan-Lim replied:
fixed!

leeyi45 commented on src/main/java/duke/ExitCommand.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
```
Might want to consider shifting some of these classes into separate packages

Jiayan-Lim replied:
fixed!

leeyi45 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,52 @@
+     */
+    public static Command parse(String input) throws DukeUnknownCommandException,
+            DukeEmptyArgumentException, DukeInvalidArgumentException {
+        String[] split = input.split(" ", 2);
```
Might want to consider case insensitive string comparisons

Jiayan-Lim replied:
fixed!

leeyi45 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,138 @@
+     * @param index task to be deleted from task list.
+     * @return message for delete command.
+     */
+    public String responseToDeleteTaskCommand(TaskList taskList, int index) {
```
You could probably put these within the commands' classes

Jiayan-Lim replied:
fixed!

shaowi commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,196 @@
+     */
+    protected void updateData(int lineNumber, int status) throws DukeIoException {
+        try {
+            List<String> allLine = Files.readAllLines(path);
```
Plural form of variable "allLine" can be considered

Jiayan-Lim replied:
Nice catch! thanks

leeyi45 commented on src/main/java/duke/Events.java:
```diff
@@ -0,0 +1,59 @@
+     *
+     * @return string representation of event task.
+     */
+    public String toString() {
```
Maybe try out String.format?

Jiayan-Lim replied:
Nice suggestion! Thanks a lot!

shaowi reviewed (commented):
Can consider grouping the files in more sub folders.

Looks good to merge!

Jiayan-Lim committed:
Duke: Add welcoming message.

When the user first download the program, it won't show welcoming
message to the screen and left it blank.

Need to print welcoming message to the screen otherwise user might think
that the program goes wrong.

Add a warm welcoming message.

Let's update the duke program so that the welcoming message can be shown
in GUI.

Jiayan-Lim committed:
Update the entry point for Duke program

Jiayan-Lim committed:
Add assertion

Jiayan-Lim committed:
Improve code quality

Within certain file, there are method with too long implementation and
weak SLAP.

Code needs to be read, understood, and modified by other developers
later on. Even if the author do not intend to pass the code to someone
else, code quality is still important because the author will become a
'stranger' to his/her own code someday.

Use refactoring and guard clause to avoid improve readability.

Examine the code and refactor to improve the code quality where
necessary.

Jiayan-Lim committed:
Merge pull request #2 from Jiayan-Lim/branch-A-Assertions

Add assertion

Jiayan-Lim committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/MarkCommand.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/UnmarkCommand.java

Jiayan-Lim committed:
Merge pull request #3 from Jiayan-Lim/branch-A-CodeQuality

Improve code quality

Jiayan-Lim committed:
Detect scheduling anomalies of event

Jiayan-Lim committed:
Shift classes to different packages

Jiayan-Lim committed:
Find command: show printout message

If find command couldn't find any matched task, it don't show anything
on the screen.

This will mislead the user that the program is not working well or some
error happen.

Show the message "Couldn't find matched task" to indicate that there is
no matched task found.

Jiayan-Lim committed:
Improve code quality based on GitHub PR reviews

Jiayan-Lim committed:
Add more automated tests

Jiayan-Lim committed:
Fix the bug

All commands now are case insensitive.
Index command(Delete, mark, unmark) now can handle the case when the
user input negative number.

Jiayan-Lim committed:
Fix the bug(Continue)

leftover from last commit.

Jiayan-Lim committed:
Improve the GUI

Jiayan-Lim committed:
Improve code quality based on IP fork's issue tracker

Jiayan-Lim committed:
Fix all capital in Event and Deadline command

Jiayan-Lim committed:
Reduce font size

Jiayan-Lim committed:
Fix the bug in detecting overlapping event

Jiayan-Lim committed:
Update README.md

Jiayan-Lim committed:
Update README.md

Jiayan-Lim committed:
Add screenshot of the product

Jiayan-Lim committed:
Update JUnit test case

Jiayan-Lim committed:
Fix bug in detecting overlapping events.

Jiayan-Lim committed:
Add exit command for user guide
